homeassistant/components/modbus/binary_sensor.py

```diff
@@ -61,10 +80,41 @@ async def async_update(self, now: datetime | None = None) -> None:
                 return
             self._lazy_errors = self._lazy_error_count
             self._attr_available = False
-            self.async_write_ha_state()
-            return
+            result_bits = None
+        else:
+            self._lazy_errors = self._lazy_error_count
+            self._attr_is_on = result.bits[0] & 1
+            self._attr_available = True
+            result_bits = result.bits
 
-        self._lazy_errors = self._lazy_error_count
-        self._attr_is_on = result.bits[0] & 1
-        self._attr_available = True
+        self.schedule_update_ha_state()
         self.async_write_ha_state()
+        for entry in self._virtuals:
+            entry.set_from_master(result_bits, self._attr_available)
+
+
+class VirtualSensor(RestoreEntity, BinarySensorEntity):
```

Comment:
We don't allow for "virtual" entities in Home Assistant.
For such cases, we offer the use of template entities instead.

Reply:
I think you misread the code, behind every virtual sensor is a physical input so templates will not work.

The idea is only to do a network read once, all sensors are then received as a bitarray and split out to the virtual sensors. Consequently we reduce the number of reads dramatically, especially when considering that reading a single bit actually returns a byte.

Reply:
but if the word “virtual” disturbes you I can call them SlaveSensor.

Reply:
Right multiple sensors from a single datapoint, now it makes more sense indeed, thanks! 👍 

I would avoid the `Virtual` naming indeed.

Reply:
Changed